### PR TITLE
[ENH] Migrate `ResNetClassifier` from `sktime-dl` to `sktime` 

### LINKED ISSUE #3351: [ENH] Migrating Estimators from sktime-dl
**Is your feature request related to a problem? Please describe.**
Related to the task of migrating estimators from [sktime-dl](https://github.com/sktime/sktime-dl) to sktime. The main goal of doing so is to have all networks/models in a single package, following similar structure and quality. 

## Steps for Migration

1. Choose one of the Classifiers or Regressors, and create an issue for it.
2. You will need to create 2 new files for any DL estimator you choose:
    a. In `sktime/sktime/classification/deep_learning`: [here](https://github.com/alan-turing-institute/sktime/tree/main/sktime/classification/deep_learning), create a new file named after the network.
   b. In `sktime/sktime/networks`: [here](https://github.com/alan-turing-institute/sktime/tree/main/sktime/networks) create a new file named after the network.

## DL Classifiers:
- [x] [CNN Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_cnn.py) #2991 #2447 
- [ ] [CNTC Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_cntc.py) #3171 
- [ ] [Encoder Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_encoder.py) #3293 
- [ ] [FCN Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_fcn.py) #3233 
- [ ] [InceptionTime Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_inceptiontime.py) #3003 
- [ ] [LSTMFCN Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_lstmfcn.py) #3292 
- [ ] [MACNN Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_macnn.py)  #3294 
- [ ] [MCDCNN Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_mcdcnn.py) #3295 
- [ ] [MCNN Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_mcnn.py)
- [x] [MLP Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_mlp.py) #3232 
- [ ] [ResNet Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_resnet.py)
- [ ] [TapNet Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_tapnet.py)
- [ ] [TLENet Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_tlenet.py)
- [ ] [Twiesn Classifier](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/classification/_twiesn.py)

## DL Regressors:

- [x] [CNN Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_cnn.py) #2902 
- [ ] [CNTC Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_cntc.py)
- [ ] [Encoder Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_encoder.py)
- [ ] [FCN Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_fcn.py)
- [ ] [InceptionTime Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_inceptiontime.py)
- [ ] [LSTM Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_lstm.py)
- [ ] [LSTMFCN Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_lstmfcn.py)
- [ ] [MCDCNN Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_mcdcnn.py)
- [ ] [MLP Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_mlp.py)
- [ ] [ResNet Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_resnet.py)
- [ ] [RNN Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_rnn.py)
- [ ] [TapNet Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_tapnet.py)
- [ ] [TLENet Regressor](https://github.com/sktime/sktime-dl/blob/master/sktime_dl/regression/_tlenet.py)

### PR DESCRIPTION
**Pull request by nilesh05apr**
#### Reference Issues/PRs
Part of #3351. See also #3365 

#### What does this implement/fix? Explain your changes.
Migrating dl estimators from sktime_dl to sktime

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?
tests

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

### CODE REVIEW

File: sktime/networks/resnet.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+__author__ = ["James Large", "Withington", "nilesh05apr"]
+
+from sktime.networks.base import BaseDeepNetwork
+from sktime.utils.validation._dependencies import (
+    _check_dl_dependencies,
+    _check_soft_dependencies,
+)
+
+_check_soft_dependencies(
```
Comment (achieveordie) [inline sktime/networks/resnet.py:12]: Any reason to have this here? I don't think you're making use of the import anywhere. This estimator can run without having `keras-self-attention` as a soft-dep.

File: sktime/networks/resnet.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    scratch with deep neural networks: A strong baseline}, author={Wang,
+    Zhiguang and Yan, Weizhong and Oates, Tim}, booktitle={2017
+    International joint conference on neural networks (IJCNN)}, pages={
+    1578--1585}, year={2017}, organization={IEEE} }
+
+    """
+
+    def __init__(self, random_state=0):
+        _check_dl_dependencies(severity="error")
+        _check_soft_dependencies(
```
Comment (achieveordie) [inline sktime/networks/resnet.py:51]: Refer to the initial point.

File: sktime/networks/resnet.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            "keras-self-attention",
+            package_import_alias={"keras-self-attention": "keras_self_attention"},
+            severity="error",
+        )
+        super(ResNetNetwork).__init__()
+        self.random_state = random_state
+
+    def build_model(self, input_shape, **kwargs):
```
Comment (achieveordie) [inline sktime/networks/resnet.py:59]: The abstract method that all estimators implement is `build_network()`. This minor reason is why tests are failing across all runners. A small typo :)

File: sktime/classification/deep_learning/resnet.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+"""Multi Layer Perceptron Network (MLP) for classification."""
```
Comment (achieveordie) [inline sktime/classification/deep_learning/resnet.py:2]: "Residual Network (ResNet) for classification" would be more accurate.

File: sktime/classification/deep_learning/resnet.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    scratch with deep neural networks: A strong baseline,
+    International joint conference on neural networks (IJCNN), 2017.
+
+    Examples
+    --------
+    >>> from sktime.classification.deep_learning.resnet import ResNetClassifier
+    >>> from sktime.datasets import load_unit_test
+    >>> X_train, y_train = load_unit_test(split="train")
+    >>> clf = ResNetClassifier(n_epochs=20, bacth_size=4) # docktest: +SKIP
+    >>> clf.fit(X_train, Y_train) # docktest: +SKIP
```
Comment (achieveordie) [inline sktime/classification/deep_learning/resnet.py:61]: `# doctest: +SKIP` to have them skipped, another small typo is causing doctest failure in the checks.

**Review by achieveordie — CHANGES_REQUESTED** [review]
Some minor typos are resulting in current failures in checks. Let's get them sorted so this estimator passes through.

**Comment by achieveordie** [comment]
You could go through the tests to see if there are any failures that I missed. LMK if you need any help. This is almost done, great work!

**Comment by nilesh05apr** [comment]
@achieveordie can you please review test results? it is failing for assertion error. Thank you

**Comment by achieveordie** [comment]
Regarding `AssertionError` in [test_fit_does_not_overwrite_hyper_params](https://github.com/sktime/sktime/actions/runs/3620372174/jobs/6117748811#step:7:1610):

An excellent way to see what went wrong is to check what the error is, here it says: 

> AssertionError: Estimator ResNetClassifier should not change or mutate  the parameter optimizer from None to <keras.optimizers.optimizer_experimental.adam.Adam object at 0x7fbc8d5f2470> during fit.

which makes sense since we don't want to change parameters passed by the user. For that purpose, we actually use `self.optimizer_` (and not `self.optimizer`) as the parameter we set in case the user doesn't provide any value. You can check out [any](https://github.com/sktime/sktime/blob/main/sktime/classification/deep_learning/cnn.py#L141) of the accepted estimators to compare your own model since the only change thing that changes is the actual implementation of the estimator, the mechanics remain similar, if not the same.

**Comment by achieveordie** [comment]
(While we are still changing things, another easy win is renaming `lr` to `learning_rate` in the same line to avoid [this](https://github.com/sktime/sktime/actions/runs/3620372174/jobs/6117749271#step:10:795) warning.)

**Review by achieveordie — APPROVED** [review]

**Pull request merged**